File: CodingBat/Java/Warmup-1/frontBack.java
/*

Given a string, return a new string where the first and last chars have been exchanged.


frontBack("code") → "eodc"
frontBack("a") → "a"
frontBack("ab") → "ba"
*/

class frontBack{
    public String frontBack(String str) {
        if (str.length()< 2) return str;
        String body = str.substring(1, str.length()-1);
        return str.charAt(str.length()-1) + body + str.charAt(0);
      }      
}

File: CodingBat/Python/Warmup-1/front_back.py
"""
Given a string, return a new string where the first and last chars have been exchanged.


front_back('code') → 'eodc'
front_back('a') → 'a'
front_back('ab') → 'ba'

@author unobatbayar

"""

def front_back(str):
  if len(str) == 1:
    return str
  elif len(str) > 1:
    array = list(str)
    array[0], array[len(str)-1] = array[len(str)-1], array[0]
    newstring = ""
    for x in array:
      newstring = newstring + x
    
    str = newstring
    return str
  else:
    return str

 